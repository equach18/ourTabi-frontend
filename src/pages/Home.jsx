import { Navigate, Link } from "react-router-dom";
import homePic from "../assets/home.jpg";

function Home({ currentUser }) {
  if (currentUser) return <Navigate to="/dashboard" />;

  return (
    <div className="min-h-[calc(100vh-4rem)] flex flex-col items-center justify-center px-6 py-12">
      <div className="flex flex-col md:flex-row items-center justify-center gap-10 max-w-6xl w-full">
        {/* Left: Content */}
        <div className="w-full max-w-xl text-center md:text-left space-y-6">
          <p className="text-sm uppercase tracking-wide text-emerald-600 font-semibold">
            Your shared journey begins here
          </p>
          <h1 className="text-4xl font-bold text-orange-500">
            Welcome to OurTabi!
          </h1>
          <p className="text-lg text-zinc-800">
            Plan your trips, invite friends, and create unforgettable travel
            experiences. Make travel planning effortless at OurTabi.
          </p>
          <div className="space-x-4">
            <Link
              to="/login"
              className="bg-slate-500 text-white px-6 py-3 rounded-full shadow-md hover:bg-slate-600 transition-transform hover:scale-105"
            >
              Login
            </Link>
            <Link
              to="/signup"
              className="bg-emerald-600 text-white px-6 py-3 rounded-full shadow-md hover:bg-emerald-700 transition-transform hover:scale-105"
            >
              Sign Up
            </Link>
          </div>
        </div>
        {/* 
        Right: Image
        <div className="w-full md:w-1/2 max-h-[580px] overflow-hidden rounded-xl">
          <img
            src={homePic}
            alt="Japanese temple"
            className="w-full h-full object-cover"
          />
        </div> */}
        <div className="relative group w-full md:w-1/2 h-[580px] rounded-xl overflow-hidden shadow-xl border border-stone-200">
          {/* Image */}
          <img
            src={homePic}
            alt="Japanese temple"
            className="w-full h-full object-cover transition duration-500 group-hover:opacity-0"
          />

          {/* Text Overlay */}
          <div className="absolute inset-0 bg-white text-zinc-700 p-6 flex flex-col justify-center items-center opacity-0 transition duration-800 group-hover:opacity-400">
            <h3 className="text-xl font-semibold text-orange-500 mb-4">
              Why OurTabi?
            </h3>
            <ul className="space-y-3 text-sm text-center">
              <li>Plan group trips easily</li>
              <li>Vote on activities</li>
              <li>Chat with trip members</li>
              <li>Explore other itineraries</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Home;
