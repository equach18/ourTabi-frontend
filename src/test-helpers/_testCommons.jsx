import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserContext from "../context/UserContext";
import TripContext from "../context/TripContext";

export const testUser1 = {
  id: 1,
  firstName: "Test1",
  lastName: "User1",
  username: "testUser1",
  profilePic: "",
};
export const testUser2 = {
  id: 2,
  firstName: "Test2",
  lastName: "User2",
  username: "testUser2",
  profilePic: "testUser2.png",
};
export const testUser3 = {
  id: 3,
  firstName: "Test2",
  lastName: "User2",
  username: "testUser3",
  profilePic: "testUser3.png",
};

export const defaultMembers = [
  {
    userId: testUser1.id,
    username: testUser1.username,
    profilePic: testUser1.profilePic,
  },
  {
    userId: testUser2.id,
    username: testUser2.username,
    profilePic: testUser2.profilePic,
  },
];

export const testActivity = {
  id: 1,
  name: "Kayaking",
  tripId: 53,
  category: "Other",
  location: "Chicago",
  description: "So much fun",
  scheduledTime: "2025-07-10T14:30:00Z",
};

export const testComment = {
  id: 1,
  text: "Test comment",
  createdAt: "2025-07-11T14:30:00Z",
  userId: 1,
};

export const testTrip = {
  id: 42,
  title: "Test trip",
  destination: "Chicago, IL",
  startDate: "2025-07-10",
  endDate: "2025-07-12",
  isPrivate: true,
};

export const testFriendsData = {
  friends: [testUser1 ],
  incomingRequests: [testUser2],
  sentRequests: [testUser3],
};

// Default mock context values
const defaultUserContext = { currentUser: testUser1 };
const defaultTripContext = {
  voteOnActivity: vi.fn(),
  votes: {},
  deleteComment: vi.fn(),
  addMember: vi.fn(),
  removeMember: vi.fn(),
  members: defaultMembers,
};

// create mock trip context
export function createMockTripContext(overrides = {}) {
  return {
    voteOnActivity: vi.fn(),
    votes: {},
    deleteComment: vi.fn(),
    members: [],
    addMember: vi.fn(),
    removeMember: vi.fn(),
    ...overrides,
  };
}

/**
 * Renders a component inside MemoryRouter
 */
export function renderWithRouter(ui) {
  return render(<MemoryRouter>{ui}</MemoryRouter>);
}

/**
 * Renders a component inside MemoryRouter + UserContext + TripContext
 */
export function renderWithAllProviders(
  ui,
  {
    userContext = defaultUserContext,
    tripContext = defaultTripContext,
    route = "/",
  } = {}
) {
  return render(
    <MemoryRouter initialEntries={[route]}>
      <UserContext.Provider value={userContext}>
        <TripContext.Provider value={tripContext}>{ui}</TripContext.Provider>
      </UserContext.Provider>
    </MemoryRouter>
  );
}

/**
 * Renders a component with just UserContext
 */
export function renderWithUserProvider(ui, { userContext } = {}) {
  return render(
    <UserContext.Provider value={userContext}>{ui}</UserContext.Provider>
  );
}
