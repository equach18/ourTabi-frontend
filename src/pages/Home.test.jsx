import { screen } from "@testing-library/react";
import Home from "./Home";
import { renderWithRouter, testUser1 } from "../test-helpers/_testCommons";

// mock img on home page
vi.mock("../assets/home.jpg", () => ({ default: "home.jpg" }));

describe("Home", () => {
  function renderHome(currentUser = null) {
    return renderWithRouter(<Home currentUser={currentUser} />);
  }

  it("matches snapshot for not logged in user", () => {
    const { asFragment } = renderHome();
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for logged in user", () => {
    const { asFragment } = renderHome(testUser1);
    expect(asFragment()).toMatchSnapshot(); 
  });

  it("redirects to /dashboard when user logs in", () => {
    renderHome(testUser1);
    expect(screen.queryByText(/welcome to ourtabi/i)).not.toBeInTheDocument();
  });

  it("renders welcome msg and correct buttons when the user is not logged in", () => {
    renderHome();

    expect(
      screen.getByText(/your shared journey begins here/i)
    ).toBeInTheDocument();
    expect(screen.getByText(/welcome to ourtabi/i)).toBeInTheDocument();
    expect(
      screen.getByText(/plan your trips, invite friends/i)
    ).toBeInTheDocument();

    expect(screen.getByRole("link", { name: /login/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /sign up/i })).toBeInTheDocument();
  });

  it("renders image and text overlay", () => {
    renderHome();

    const img = screen.getByAltText(/japanese temple/i);
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "home.jpg");

    expect(screen.getByText(/why ourtabi/i)).toBeInTheDocument();
    expect(screen.getByText(/plan group trips easily/i)).toBeInTheDocument();
  });
});
