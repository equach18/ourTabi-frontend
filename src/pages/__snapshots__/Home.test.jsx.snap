// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > matches snapshot for logged in user 1`] = `<DocumentFragment />`;

exports[`Home > matches snapshot for not logged in user 1`] = `
<DocumentFragment>
  <div
    class="min-h-[calc(100vh-4rem)] flex flex-col items-center justify-center px-6 py-12"
  >
    <div
      class="flex flex-col md:flex-row items-center justify-center gap-10 max-w-6xl w-full"
    >
      <div
        class="w-full max-w-xl text-center md:text-left space-y-6"
      >
        <p
          class="text-sm uppercase tracking-wide text-emerald-600 font-semibold"
        >
          Your shared journey begins here
        </p>
        <h1
          class="text-4xl font-bold text-orange-500"
        >
          Welcome to OurTabi!
        </h1>
        <p
          class="text-lg text-zinc-800"
        >
          Plan your trips, invite friends, and create unforgettable travel experiences. Make travel planning effortless at OurTabi.
        </p>
        <div
          class="space-x-4"
        >
          <a
            class="bg-slate-500 text-white px-6 py-3 rounded-full shadow-md hover:bg-slate-600 transition-transform hover:scale-105"
            data-discover="true"
            href="/login"
          >
            Login
          </a>
          <a
            class="bg-emerald-600 text-white px-6 py-3 rounded-full shadow-md hover:bg-emerald-700 transition-transform hover:scale-105"
            data-discover="true"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
      <div
        class="relative group w-full md:w-1/2 h-[580px] rounded-xl overflow-hidden shadow-xl border border-stone-200"
      >
        <img
          alt="Japanese temple"
          class="w-full h-full object-cover transition duration-500 group-hover:opacity-0"
          src="home.jpg"
        />
        <div
          class="absolute inset-0 bg-white text-zinc-700 p-6 flex flex-col justify-center items-center opacity-0 transition duration-800 group-hover:opacity-400"
        >
          <h3
            class="text-xl font-semibold text-orange-500 mb-4"
          >
            Why OurTabi?
          </h3>
          <ul
            class="space-y-3 text-sm text-center"
          >
            <li>
              Plan group trips easily
            </li>
            <li>
              Vote on activities
            </li>
            <li>
              Chat with trip members
            </li>
            <li>
              Explore other itineraries
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
